`cudacodec::VideoWriter` add flag for writing full range luma chroma video

cudawarped opened the pull request:
Currently the meta data in all encoded video's is not updated to reflect the range (full of limited as defined in Annex E of the ITU-T Specification) of encoded luma/chroma values.  With all video's indicating that they use the limited range.  This means that when a full range YUV source is encoded the decoder will incorrectly decode it to a limited range resulting in poor color reproduction.

This PR correctly sets this meta data field for all encoded video's.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

asmorkalov commented on modules/cudacodec/src/video_writer.cpp:
```diff
@@ -342,10 +342,20 @@ void VideoWriterImpl::InitializeEncoder(const GUID codec, const double fps)
+            initializeParams.encodeConfig->encodeCodecConfig.hevcConfig.hevcVUIParameters.videoFullRangeFlag = 1;
+            initializeParams.encodeConfig->encodeCodecConfig.hevcConfig.hevcVUIParameters.videoSignalTypePresentFlag = 1;
+        }
+    }
```
I propose to print warning, if full range luma is used with other codec.

cudawarped replied:
Good call especially as the Blackwell will add support for AV1 encoding.

cudawarped commented:
@asmorkalov I've added a warning when the codec is not h264 or h265.

**Note:** Because I can't test AV1 encoding (I don't have access to a card with 8th Gen NVDEC) the cudacodec module does not currently support AV1 encoding.  This is why I haven't also added AV1 support for the `videoFullRangeFlag` (`NV_ENC_CONFIG::encodeConfig->encodeCodecConfig.av1Config.colorRange`) flag in this PR.

asmorkalov commented:
You actually can submit the PR  with GF 4x support. I have a device for test.

cudawarped commented:
That's good to know although I'll need access to one first to test everything locally first before submitting a PR.

asmorkalov commented:
@cudawarped There is conflict with the previous PR please rebase. Also `videoFullRangeFlag` looks ugly for me. It's bool and it's obviously flag. I propose to rename it to something like `encodeVideoFullRange`.

cudawarped commented:
@asmorkalov Correct, I copied the name from the SDK where its a flag (`unsigned char video_full_range_flag` in NVDEC and `uint32_t  videoFullRangeFlag` in NVENC).  Additionaly as I mentioned above this is refered to as `uint32_t colorRange` for AV1 encoding so a differently named bool here would make sense.  I tried to keep the naming the same as the SDK for transparency but I can change this if its less confusing?

For consistency should I also rename it in `VideoReader` to `videoFullRangeSource` or similar?

asmorkalov commented:
`videoFullRangeSource` looks good.

asmorkalov commented:
Oh, got the issue. Let it be videoFullRangeFlag everywhere to not break code and confuse users.
